src/encoder/ffmpeg/codec.cpp

```diff
@@ -4,11 +4,84 @@ extern "C" {
 }
 #include "disable_warnings_pop.hpp"
 
-#include "codec.hpp"
+#include <memory>
+#include <optional>
 
 #include "encoder/convert.hpp"
 #include "common/time.hpp"
+#include "codec.hpp"
+
+namespace {
+  static const int    buf_size = 250;
+  static const int    prefix_length = 9;
+  static const char   log_prefix[prefix_length + 1] = "[ffmpeg] "; // +1 for /0
+  static std::optional<shar::Logger> cb_logger;
+
+  static void avlog_callback(void * ptr, int level, const char * fmt, va_list args) {
+    if (cb_logger) {
+      char buf[buf_size];
+      std::memcpy(buf, log_prefix, prefix_length);
+      vsprintf(buf + prefix_length, fmt, args);
+      switch (level) {
+      case AV_LOG_TRACE:
+        cb_logger->trace(buf);
+      case AV_LOG_DEBUG:
+      case AV_LOG_VERBOSE:
+        cb_logger->debug(buf);
+        break;
+      case AV_LOG_INFO:
+        cb_logger->info(buf);
+        break;
+      case AV_LOG_WARNING:
+        cb_logger->warning(buf);
+        break;
+      case AV_LOG_ERROR:
+        cb_logger->error(buf);
+        break;
+      case AV_LOG_FATAL:
+      case AV_LOG_PANIC:
+        cb_logger->critical(buf);
+        break;
+      default:
+        break;
+      }
+    }
+  }
 
+  static void setup_logging(const shar::OptionsPtr& config, shar::Logger& logger) {
```

Comment:
There should be check that `encoder_loglvl` >= `loglvl`, otherwise all extra messages that `envoder_loglvl` enables will be filtered